src/App.jsx

```diff
@@ -4,19 +4,36 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
 import { Nav } from "./components";
 import ArticlePage from "./components/ArticlePage";
 import { Stack } from "@mui/material";
+import { useMediaQuery } from "@mui/material";
+import { useMemo } from "react";
+import { createTheme } from "@mui/material";
+import { ThemeProvider } from "@mui/material";
 
 function App() {
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const theme = useMemo(
+    () =>
+      createTheme({
+        palette: {
+          mode: prefersDarkMode ? "dark" : "light",
+        },
+      }),
+    [prefersDarkMode]
+  );
+
   return (
     <>
-      <BrowserRouter>
-        <Stack alignItems={"center"}>
-          <Nav />
-          <Routes>
-            <Route path="/" element={<ArticleList />} />
-            <Route path="/article/:articleId" element={<ArticlePage />} />
-          </Routes>
-        </Stack>
-      </BrowserRouter>
+      <ThemeProvider theme={theme}>
+        <BrowserRouter>
+          <Stack alignItems={"center"}>
+            <Nav />
+            <Routes>
+              <Route path="/" element={<ArticleList />} />
+              <Route path="/article/:articleId" element={<ArticlePage />} />
```

Comment:
This might be better as articles/articleId as it comes from multiple articles to one single article